bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -97,26 +97,44 @@ public void propertyChange(PropertyChangeEvent evt) {
                 mm.addMarker(false, this, Severity.Warning, msg, v, a);
             }
 
-			// Function to create markers that report solver intervals.
-			final BiFunction<String, String, Option<Marker>> createSolverMarker =
-				(cacheName, prefix) -> {
+            final Supplier<Union<Interval>> wholeNight = () -> new Union<>(new Interval(v.getSchedule().getStart(), v.getSchedule().getEnd()));
+
+			final Predicate<String> contributes =
+				(cacheName) -> {
 					final Map<Obs, Union<Interval>> c = v.getSchedule().getCache(cacheName);
-					final Union<Interval>       valid = c.getOrDefault(a.getObs(), new Union<>());
+					final Union<Interval> u = c.getOrDefault(a.getObs(), new Union<>());
 
-					final Union<Interval>     invalid = new Union<>(new Interval(v.getSchedule().getStart(), v.getSchedule().getEnd()));
-					invalid.remove(valid);
+					final Union<Interval> r = wholeNight.get();
+					r.remove(u);
+					return !r.isEmpty();
+				};
 
+			// Function to create markers that report solver intervals.
+			final Function<String, Option<Marker>> createSolverMarker =
+				(prefix) -> {
+					final Map<Obs, Union<Interval>> c = v.getSchedule().getCache(Variant.CONSTRAINED_UNION_CACHE);
+					final Union<Interval>       valid = c.getOrDefault(a.getObs(), new Union<>());
+					final Union<Interval>     invalid = wholeNight.get();
+					invalid.remove(valid);
 					final ImList<Interval>         is = DefaultImList.create(valid.getIntervals());
 
 					if (is.isEmpty() || invalid.isEmpty()) {
 						return None.instance();
 					} else {
-						return new Some<>(new Marker(false, this, Severity.Notice, prefix, new Some<>(valid), v, a));
+					    final List<String> attrs = new ArrayList<>();
+						if (contributes.test(Variant.VISIBLE_UNION_CACHE)) { attrs.add("airmass"); }
+					    if (contributes.test(Variant.DARK_UNION_CACHE))    { attrs.add("background"); }
+						if (contributes.test(Variant.TIMING_UNION_CACHE))  { attrs.add("timing window"); }
+
+                        final String s = attrs.isEmpty() ?
+							"" : DefaultImList.create(attrs).mkString(" (", ", ", ")");
+
+						return new Some<>(new Marker(false, this, Severity.Notice, prefix + s, new Some<>(valid), v, a));
```

Comment:
This will leave the attributes as part of the prefix instead of appended after the formatted time, but to achieve that we'd need to create a template message such that the timestamp can appear anywhere in the template.